src/Kernel32/Kernel32.Helpers.cs

```diff
@@ -256,6 +256,172 @@ public static ArraySegment<byte> ReadFile(SafeObjectHandle hFile, int nNumberOfB
             return new ArraySegment<byte>(buffer, 0, bytesRead);
         }
 
+        /// <summary>
+        /// Retrieves information about the specified process
+        /// </summary>
+        /// <param name="hProcess">
+        /// A handle to the process. This handle must have the <see cref="ProcessAccess.PROCESS_SET_INFORMATION"/> access right.
+        /// For more information, see <a href="https://docs.microsoft.com/en-us/windows/desktop/ProcThread/process-security-and-access-rights">Process Security and Access Rights</a>
+        /// </param>
+        /// <param name="ProcessInformationClass">
+        /// A member of the <see cref="PROCESS_INFORMATION_CLASS"/> enumeration specifying the kind of information to retrieve.
+        /// </param>
+        /// <param name="ProcessInformation">
+        /// Pointer to an object to receive the type of information specified by the <paramref name="ProcessInformationClass"/> parameter
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        ///
+        /// If the function fails, the return value is zero.To get extended error information, call <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <remarks>
+        /// In most situations, the size of payload in <paramref name="ProcessInformation"/> can be correctly inferred based on
+        /// the value of <paramref name="ProcessInformationClass"/>. This helper is intended to make it simple to call into
+        /// <see cref="GetProcessInformation(SafeObjectHandle, PROCESS_INFORMATION_CLASS, void*, uint)"/>.
+        /// </remarks>
+#pragma warning disable IDE1006 // Naming Styles
+        public static unsafe bool GetProcessInformation(
+            SafeObjectHandle hProcess,
+            PROCESS_INFORMATION_CLASS ProcessInformationClass,
+            void* ProcessInformation)
+#pragma warning restore IDE1006 // Naming Styles
+        {
+            uint processInformationSize = 0;
+
+            switch (ProcessInformationClass)
+            {
+                case PROCESS_INFORMATION_CLASS.ProcessMemoryPriority:
+                    processInformationSize = (uint)sizeof(MEMORY_PRIORITY_INFORMATION);
+                    break;
+                case PROCESS_INFORMATION_CLASS.ProcessMemoryExhaustionInfo:
+                    // PROCESS_MEMORY_EXHAUSTION_INFO is non-blittable - it has UIntPtr members; use Marshal.SizeOf
+                    // to compute struct-size instead of `sizeof`
```

Comment:
Eh, what now? `UIntPtr` is a blittable type I thought. If not, then my #493 PR may have gone too far.

Reply:
I thought I'd quickly respond to just this one. 

https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types

I had trouble applying `sizeof` to `struct` a with `IntPtr`/`UIntPtr` fields. I'll check and report back the exact error I was getting shortly.

Reply:
```
sizeof(IntPtr)
(1,1): error CS0233: 'IntPtr' does not have a predefined size, therefore sizeof can only be used in an unsafe context
```

I let myself get mislead by that. Sorry about the confusion - will fix.

Reply:
Very strange that the doc you found claims that `char` and `bool` are non-blittable.

Reply:
Initially I thought that had to be wrong, but upon more thinking I suspect that it might be correct after all. 

bool and char might very be blittable in a strict sense, but in a practical sense bool <-> BOOL mapping requires MarshalAs(UnmanagedType.Bool) marshaling, and char <-> [W]CHAR marshaling requires ANSI/UNICODE conversions sometimes, which might make these types non-blittable in a practical sense.

Reply:
Ya, but thank goodness in a strict sense they don't violate blittability of a struct.